python/src/iceberg/catalog/__init__.py

```diff
@@ -0,0 +1,16 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
```

Comment:
I thought we were planning on flattening the structure instead of having package here?

I understand wanting to set up the basic structure, but I'd probably avoid creating so many empty `__init__.py` files in this commit.

Reply:
The `__init__.py` files are required for python to see them as packages.  I believe we were planning on having this package.

Reply:
> I thought we were planning on flattening the structure instead of having package here?
I think a flat structure makes the usage look cleaner (i.e. `from iceberg import Catalog` instead of `from iceberg.catalog import Catalog`). On the flip side, developing for the library feels more manageable when everything is grouped into single-responsibility packages. I think we can get the best of both worlds by organizing the directory/package structure to optimize for the latter, and then defining an `__all__` array in the root `__init__.py` that bubbles up everything the user needs to a flat API at the top-level. How does that sound?

Reply:
I also thought we decided to flatten this to a `catalog.py` at top level. Depending on the dependency management part, we might be able to have all the catalog implementations all in that file, or if that is not possible, then we can do something like `from iceberg.aws import GlueCatalog` for plugin catalogs in separated files like `aws.py`, `hive.py`, `jdbc.py`... It feels a bit awkward to have things like `table_operation`, `partition_spec` also under the `catalog` folder.

Reply:
I am thinking if we keep the package and keep different catalog implementations within it, then users can use
```
from iceberg.catalog import hive
from iceberg.catalog import glue
...
```
So all table spec core classes are in the top level and then we put pluggable classes to the packages, `catalog`, `io`. 
Additionally, we can define them in `__init__.py` to bubble up if needed.